android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayAvailabilityChecker.kt

```diff
@@ -5,40 +5,52 @@ import androidx.fragment.app.FragmentActivity
 import com.adyen.checkout.components.core.CheckoutConfiguration
 import com.adyen.checkout.components.core.ComponentAvailableCallback
 import com.adyen.checkout.components.core.PaymentMethod
+import com.adyen.checkout.googlepay.GooglePayButtonParameters
 import com.adyen.checkout.googlepay.GooglePayComponent
 
 class GooglePayAvailabilityChecker(
     private val activity: FragmentActivity,
-    private val googlePayComponent: BaseGooglePayComponentWrapper?,
-    private val googlePaySetupCallback: (Result<InstantPaymentSetupResultDTO>) -> Unit,
 ) : ComponentAvailableCallback {
+    private var googlePayButtonParameters: GooglePayButtonParameters? = null
+    private var googlePaySetupCallback: ((Result<InstantPaymentSetupResultDTO>) -> Unit)? = null
+
     override fun onAvailabilityResult(
         isAvailable: Boolean,
         paymentMethod: PaymentMethod
     ) {
         if (isAvailable) {
-            googlePaySetupCallback(
+            googlePaySetupCallback?.invoke(
                 Result.success(
                     InstantPaymentSetupResultDTO(
                         InstantPaymentType.GOOGLEPAY,
                         true,
-                        googlePayComponent?.googlePayComponent?.getGooglePayButtonParameters()
-                            ?.allowedPaymentMethods.orEmpty()
+                        googlePayButtonParameters?.allowedPaymentMethods.orEmpty()
                     )
                 )
             )
         } else {
-            googlePaySetupCallback(Result.failure(Exception("Google pay not available")))
+            googlePaySetupCallback?.invoke(Result.failure(Exception("Google pay not available")))
         }
     }
 
     fun checkGooglePayAvailability(
```

Comment:
Minor:

How about just "checkAvailability" and "onResult" ? I would not expect `GooglePayAvailabilityChecker` to be able to check availability of something else but GooglePay

Reply:
After a second thought, using a dedicated class for only one method is maybe too much. The manager can implement the onAvailabilityResult and that would make the GooglePayAvailabilityChecker completely redundant.